src/cache/cache_las.c

```diff
@@ -1423,10 +1433,50 @@ __wt_find_lookaside_upd(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDAT
         upd->durable_ts = durable_timestamp;
         upd->start_ts = las_timestamp;
         upd->prepare_state = prepare_state;
-        /*
-         * Mark this update as external and to be discarded when not needed.
-         */
-        upd->ext = 1;
+
+        if (prepare_state == WT_PREPARE_INPROGRESS) {
+            WT_ASSERT(session, !modify);
+            WT_ASSERT(session, cbt->compare == 0);
+            upd_entry = NULL;
+            switch (page->type) {
+            case WT_PAGE_COL_FIX:
+            case WT_PAGE_COL_VAR:
+                WT_ASSERT(session, cbt->ins != NULL);
+                upd_entry = &cbt->ins->upd;
+                break;
+            case WT_PAGE_ROW_LEAF:
+                upd_entry =
+                  cbt->ins == NULL ? &page->modify->mod_row_update[cbt->slot] : &cbt->ins->upd;
+                break;
+            }
+
+            WT_ASSERT(session, upd_entry != NULL);
```

Comment:
My initial design was to use `__wt_row_modify` and `__wt_col_modify` for this. However I felt that those functions were getting a bit overloaded with different cases so I decided to just inline the relevant code here.

I think that this is a pretty specific case without too many exceptions (famous last words) so this seemed reasonable to me. Let me know if you disagree with what I've done here.

Reply:
I suggest getting Keith's opinion on this.

Reply:
@keithbostic Ok, I'm using `__wt_row_modify` and `__wt_col_modify`. Let me know how this looks.